### PR TITLE
[SIG-3623] Reset dynamic extra questions state

### PR DESCRIPTION
**Pull request by MauritsR**
### Changes

- Fixes state handling for extra questions that are dynamically added/removed

The visibility of some extra questions is dependent other questions. Previously, answers from questions that were dynamically removed (i.e. `isVisible` was set to `false`) were kept in the incident state. This caused these answers to still show up on the summary page.

### CODE REVIEW

File: src/signals/incident/components/IncidentForm/index.test.js [hunk 1]
```diff
@@ -300,4 +302,53 @@ describe('<IncidentForm />', () => {
       expect(next).not.toHaveBeenCalled();
     });
   });
+
+  describe('Extra questions', () => {
+    it('removes extra question data when the question is not visible anymore', () => {
```
Comment (aditudorache) [inline src/signals/incident/components/IncidentForm/index.test.js:307]: Can we write the tests using the @testing-libraty?

**Review by aditudorache — CHANGES_REQUESTED** [review]
Not sure if out of scope, but would be nice to convert the IncidentForm component to a functional component.

**Comment by MauritsR** [comment]
> Not sure if out of scope, but would be nice to convert the IncidentForm component to a functional component.

Out of scope for this story, but definitely something we should pick up!

**Review by aditudorache — APPROVED** [review]
Great job!

**Pull request merged**